### PR TITLE
DTFS2-5474 - GEF - ChangedToIssued facilities to unissued again

### PR DESCRIPTION
**Pull request by Zainzzkk**
# Changes made

* added ability to change issued facility to unissued if changedToIssued
* removes dates and updates db
* add to facilities model so it changes issueDate to null and does not wipe monthsOfCover if changing to issued post submission
* updated tests
* added e2e test

### CODE REVIEW

File: gef-ui/server/controllers/unissued-facilities/index.js [hunk 1]
```diff
@@ -101,6 +102,30 @@ const changeUnissuedFacilityPreview = async (req, res) => {
...
 };
 
+// when changing unissued facility from application preview page
+// renders about-facility change page for unissued facilities
+const changeIssuedToUnissuedFacility = async (req, res) => {
+  const { params, query } = req;
+  const { dealId, facilityId } = params;
+  let { facilityType } = query;
+  facilityType = facilityType || FACILITY_TYPE.CASH;
+  const facilityTypeString = FACILITY_TYPE[facilityType?.toUpperCase()].toLowerCase();
```
Comment (abhi-markan) [inline gef-ui/server/controllers/unissued-facilities/index.js:112]: Should this be a helper function?

File: gef-ui/server/controllers/unissued-facilities/index.js [hunk 2]
```diff
@@ -270,11 +299,82 @@ const postChangeUnissuedFacilityPreview = async (req, res) => {
...
+      facilityType: facilityTypeString,
+      errors: validationErrorHandler(hasBeenIssuedErrors),
+      dealId,
+    });
+  }
+
+  try {
+    if (body.hasBeenIssued === 'false') {
+      const { details } = await api.getFacility(facilityId);
+
```
Comment (abhi-markan) [inline gef-ui/server/controllers/unissued-facilities/index.js:339]: Please validate whether `details` object is not null.

File: gef-ui/server/utils/helpers.js [hunk 3]
```diff
@@ -249,13 +259,15 @@ const summaryItemsConditions = (summaryItemsObj) => {
   const value = typeof details[item.id] === 'number' || typeof details[item.id] === 'boolean' ? details[item.id].toString() : details[item.id];
   const isCoverStartOnSubmission = id === 'coverStartDate' && shouldCoverStartOnSubmission;
   // column keys to display change if facility has been changed to issued
-  const changedToIssueShow = id === 'name' || id === 'coverStartDate' || id === 'coverEndDate' || id === 'issueDate';
+  const changedToIssueShow = id === 'name' || id === 'coverStartDate' || id === 'coverEndDate' || id === 'issueDate' || id === 'hasBeenIssued';
```
Comment (abhi-markan) [inline gef-ui/server/utils/helpers.js:262]: Compound condition complexity is over 3, should be a helper function in my opinion.

**Review by abhi-markan — CHANGES_REQUESTED** [review]

File: gef-ui/server/controllers/unissued-facilities/index.js (hunk 1 above)
Comment (Zainzzkk) [reply gef-ui/server/controllers/unissued-facilities/index.js:112]: this is exactly how it is currently for facilities currently.  Only difference is that I have taken out the status field

File: gef-ui/server/utils/helpers.js (hunk 3 above)
Comment (Zainzzkk) [reply gef-ui/server/utils/helpers.js:262]: fixed

File: gef-ui/server/controllers/unissued-facilities/index.js (hunk 1 above)
Comment (abhi-markan) [reply gef-ui/server/controllers/unissued-facilities/index.js:112]: Lets chat about to it on Monday, after stand up.

Comment (Zainzzkk) [reply gef-ui/server/controllers/unissued-facilities/index.js:112]: done

File: gef-ui/server/controllers/unissued-facilities/index.js [hunk 4]
```diff
@@ -270,11 +299,82 @@ const postChangeUnissuedFacilityPreview = async (req, res) => {
...
+  const {
+    body, params, query, session,
+  } = req;
+  const { dealId, facilityId } = params;
+  const { user } = session;
+  const { _id: editorId } = user;
+  let { facilityType } = query;
+  const hasBeenIssuedErrors = [];
+  facilityType = facilityType || FACILITY_TYPE.CASH;
+  const facilityTypeString = FACILITY_TYPE[facilityType?.toUpperCase()].toLowerCase();
```
[the remark this answers is not in the document]
Comment (Zainzzkk) [reply gef-ui/server/controllers/unissued-facilities/index.js:320]: done

File: gef-ui/server/controllers/unissued-facilities/index.js (hunk 1 above)
Comment (IgnatG) [reply gef-ui/server/controllers/unissued-facilities/index.js:112]: @Zainzzkk  I can't see the changes

File: gef-ui/server/controllers/unissued-facilities/index.js (hunk 2 above)
Comment (Zainzzkk) [reply gef-ui/server/controllers/unissued-facilities/index.js]: done

File: gef-ui/server/controllers/unissued-facilities/index.js (hunk 1 above)
Comment (IgnatG) [reply gef-ui/server/controllers/unissued-facilities/index.js:113]: I still can't see the changes for this @Zainzzkk  :)

File: gef-ui/server/utils/facility-helpers.js [hunk 5]
```diff
@@ -195,6 +195,8 @@ const issuedFacilityConfirmation = (application) => {
   && (submissionType === CONSTANTS.DEAL_SUBMISSION_TYPE.MIN || submissionType === CONSTANTS.DEAL_SUBMISSION_TYPE.AIN);
 };
 
+const facilityTypeStringGenerator = (facilityType) => CONSTANTS.FACILITY_TYPE[facilityType?.toUpperCase()].toLowerCase();
```
Comment (abhi-markan) [inline gef-ui/server/utils/facility-helpers.js:198]:
``` const facilityTypeStringGenerator = (facilityType) => {
    if (facilityType) {
        const type = CONSTANTS.FACILITY_TYPE[facilityType.toUpperCase()];
        return type.toLowerCase();
    }
    return null;
} ```

**Review by abhi-markan — CHANGES_REQUESTED** [review]
Minor change please

File: gef-ui/server/utils/facility-helpers.js (hunk 5 above)
Comment (Zainzzkk) [reply gef-ui/server/utils/facility-helpers.js]: done

**Review by abhi-markan — APPROVED** [review]

**Pull request merged**